src/api/call-api.ts

```diff
@@ -15,21 +15,15 @@ export const callApi = async ({
 }: CallApiOptions): Promise<node.LogsData<ethers.BigNumber | null>> => {
   const configOis = oises.find((o) => o.title === oisTitle)!;
   const configEndpoint = configOis.endpoints.find((e) => e.name === endpointName)!;
-  const reservedParameters = node.adapters.http.parameters.getReservedParameters(
-    configEndpoint,
-    apiCallParameters || {}
-  );
+  const reservedParameters = node.adapters.http.parameters.getReservedParameters(configEndpoint, apiCallParameters);
   if (!reservedParameters._type) {
     const message = `reserved parameter '_type' is missing for endpoint: ${endpointName}`;
     const log = node.logger.pend('ERROR', message);
     return [[log], null];
   }
 
   // Remove reserved parameters
-  const sanitizedParameters: adapter.Parameters = node.utils.removeKeys(
-    apiCallParameters || {},
-    ois.RESERVED_PARAMETERS
-  );
+  const sanitizedParameters: adapter.Parameters = node.utils.removeKeys(apiCallParameters, ois.RESERVED_PARAMETERS);
```

Comment:
Is `apiCallParameters` always going to exist? I see the `|| {}` default has been removed

Reply:
good question 🤔 

let's see, `abi.decode` doesn't return null or undefined and worst case is empty object: https://github.com/api3dao/airkeeper/blob/bec210-refactor-airkeeper/src/handlers/psp.ts#L168

`CallApiOptions` interface also doesn't allow null or undefined.

I guess that if endpoint doesn't use any parameter then this argument should be an empty object making the `|| {}` a bit redundant.

Do you still think is worth adding that extra safety measure?

Reply:
It depends on what behaviour we want here. If something has gone wrong (that shouldn't have gone wrong), then we should probably let this throw/crash?